secedgar/tests/filings/test_filings.py

```diff
@@ -10,18 +10,37 @@
 from secedgar.utils.exceptions import FilingTypeError, EDGARQueryError
 
 
-class MockSingleCIKNotFound:
+class MockResponse:
+    def __init__(self, datapath_args=[], status_code=200, file_read_args='r', text=None, *args):
+        self.status_code = status_code
+        if text is not None:
+            self.text = text
+        else:
+            with open(datapath(*datapath_args), file_read_args) as f:
+                self.text = f.read()
+
+
+class MockSingleCIKNotFound(MockResponse):
     def __init__(self, *args):
-        self.status_code = 200
-        with open(datapath("CIK", "cik_not_found.html"), 'rb') as f:
-            self.text = f.read()
+        super().__init__(["CIK", "cik_not_found.html"], 'rb')
 
 
-class MockSingleCIKFiling:
+class MockSingleCIKFiling(MockResponse):
     def __init__(self, *args):
-        self.status_code = 200
-        with open(datapath('filings', 'aapl_10q_filings.xml')) as f:
-            self.text = f.read()
+        super().__init__(['filings', 'aapl_10q_filings.xml'])
+
+
+class MockSingleCIKFilingLimitedResponses:
```

Comment:
Need to renew called count for each CIK

Reply:
Will stick to using single CIK lookup for testing instead. If multiple CIK lookup becomes an issue, then this will need to be taken care of.